test/integration/full/controls/shared-vpc.rb

```diff
@@ -88,28 +88,41 @@
     end
   end
 
-  describe command("gcloud beta compute networks subnets get-iam-policy #{shared_vpc_subnet_name} --region #{shared_vpc_subnet_region} --project #{shared_vpc} --format=json") do
-    its('exit_status') { should eq 0 }
-    its('stderr') { should eq '' }
-
-    let(:bindings) do
-      if subject.exit_status == 0
-        JSON.parse(subject.stdout, symbolize_names: true)[:bindings]
-      else
-        []
+  SHARED_VPC_SUBNETS_DATA_SET = [
+    {
+      "name" => shared_vpc_subnet_name_01,
+      "region" => shared_vpc_subnet_region_01
+    },
+    {
+      "name" => shared_vpc_subnet_name_01,
+      "region" => shared_vpc_subnet_region_01
+    }
+  ]
+
+  SHARED_VPC_SUBNETS_DATA_SET.each do |subnet|
```

Comment:
Test cases must not be generated through metaprogramming; this practice can lead to false positives if the logic driving the generation happens to change.

Reply:
I agree, but how would you minimize code duplication in this case? basically we need to do same code against 2 different networks

Reply:
Reducing code duplication in test cases is less of a priority than creating clear test cases. Duplicating logic in this scenario is totally acceptable.